flexget/components/sites/sites/filelist_api.py

```diff
@@ -0,0 +1,166 @@
+            'internal': {'type': 'boolean', 'default': False},
+            'moderated': {'type': 'boolean', 'default': False},
+            'freeleech': {'type': 'boolean', 'default': False},
+            'doubleup': {'type': 'boolean', 'default': False},
+        },
+        'required': ['username', 'passkey'],
+        'additionalProperties': False,
+    }
+
+    def get(self, url, params):
+
+        try:
+            response = requests.get(url, params=params, raise_status=False)
+        except RequestException as e:
+            raise plugin.PluginError(f'FileList request failed badly! {e}')
+
+        if not response.ok:
+            http_status = HTTPStatus(response.status_code)
+            error_message = response.json().get('error', http_status.description)
+
+            raise plugin.PluginError(f'FileList request failed; err code: {http_status}; err msg: `{error_message}`')
+
+        return response
+
+    @plugin.internet(logger)
+    def search(self, task, entry, config):
+        """
+            Search for entries on FileList
+        """
+        entries = []
+
+        # TODO remove later
+        logger.debug(entry)
+
+        # mandatory params
+        params = {'username': config['username'], 'passkey': config['passkey'], 'action': 'search-torrents'}
+
+        # category
+        if config.get('category'):
+            params['category'] = (
+                ','.join(self.valid_categories[cat] for cat in config['category'])
+                if isinstance(config.get('category'), list)
+                else self.valid_categories[config.get('category')]
+            )
+
+        # extras: internal release, moderated torrent, freeleech
+        for extra_name, extra_value in self.valid_extras.items():
+            if config.get(extra_name):
+                params[extra_name] = extra_value
+
+        # set season/episode if series
+        if entry.get('series_episode'):
+            params['episode'] = entry.get('series_episode')
+        if entry.get('series_season'):
+            params['season'] = entry.get('series_season')
+
+        for search_title in entry.get('search_strings', [entry.get('title')]):
+
+            if entry.get('imdb_id'):
+                params['type'] = 'imdb'
+                params['query'] = entry.get('imdb_id')
+                params['name'] = search_title
+            else:
+                params['type'] = 'name'
+                params['query'] = search_title
+
+            # TODO remove later
+            logger.debug('get params: {}', params)
+
+            # get them results
+            try:
+                response = self.get(self.api_url, params)
+                # TODO remove later
+                logger.debug('Requesting: {}', response.url)
+            except RequestException as e:
+                raise plugin.PluginError('FileList request failed badly! {}', e)
+
+            results = response.json()
+            if not results:
```

Comment:
will this be an empty list if there are no results?

Reply:
as far as i remember if i don't get any result it will be an empty list, hopefully they don't change the behavior 😄